src/components/legend/draw.js

```diff
@@ -465,10 +465,10 @@ function clickOrDoubleClick(gd, legend, legendItem, numClicks, evt) {
         evtData.label = legendItem.datum()[0].label;
     }
 
-    var clickVal = Events.triggerHandler(gd, 'plotly_legendclick', evtData);
-    if(clickVal === false) return;
-
     if(numClicks === 1) {
+        var clickVal = Events.triggerHandler(gd, 'plotly_legendclick', evtData);
+        if(clickVal === false) return;
```

Comment:
I think we still want `plotly_legendclick` events to fire on `numClicks>1`, just prevent the `handleClick` call from being scheduled. So perhaps leave the first of these two lines outside the `if`, only move the second one inside?

Then I guess we can ask whether returning `false` from the `plotly_legendclick` event should also prevent the doubleclick internal behavior, or if you also need to return `false` from `plotly_doubleclick`? I guess to minimize disruption of existing code we should still have the first `false` prevent all internal behavior, so the `dblClickVal !== false` below will need to be `(dblClickVal !== false) && (clickVal !== false)`

Reply:
The overall issue is just to have the ability to use double click handlers even when you want to cancel the default single click handler. 

I don't really feel comfortable expressing a strong opinion here as I'm not a maintainer and I'm not familiar with the code base, but I think it could be considered to not fire the plotly_legendclick event at all if you have more than 1 click. In plain JS this is totally fine because there's only one 'click' event and you have to add your own logic within your handler based on the event details to decide what exactly you're trying to handle (single click, double, duration, etc). In this case you're explicitly splitting it into two different events so the fact that they are then intertwined together and not distinct is very not intuitive, at least for me.

In any case, I'm fine with minimizing the changes if you think it could break some expectations/patterns. Are you going to edit this PR yourself or do you want me to try to interpret your comments and take a stab at it?

Reply:
You can update the PR 😎

My goals are (1) minimize the chance a change will break existing behavior that users depend on - hence my comment re: returning `false` from the first preventing all internal handlers, and (2) behave as consistently as possible with built-in events - and AFAICT plain `click` and `dblclick` do what I’m proposing, the `click` fires even if the same click is about to trigger a `dblclick`

Reply:
So I tweaked it a bit, but the check for return false still has to be within the if statement otherwise a canceled single click will never allow the double click to fire which was the whole point of this...

Reply:
Better! Still I think a `false` from the `plotly_legendclick` should prevent the internal double click behavior as well, for backward compatibility. That's the last piece of my first comment:

> the `dblClickVal !== false` below will need to be `(dblClickVal !== false) && (clickVal !== false)`

Would this cause any problems?

Reply:
Ah okay, I didn't see what you were saying at first but now I get it and it makes sense as both of the default behaviors are similar/related in this case. My current desired scenario is to disable single click but allow custom double click so I can live with that.

Reply:
Yep, I'm happy with it now, thanks! I know @archmoj will want to see a [draftlog](https://github.com/plotly/plotly.js/tree/master/draftlogs) `6727_fix.md` before we merge, anything else?

Reply:
Ok I just added a draftlog file for this. Anything else?

Reply:
@andrej-vasilj Thanks very much for the PR. 
Please add a test in `test/jasmine/tests/legend_test.js` to lock down the bug.

Reply:
No prob, I just added a simple test that covers this use-case